src/compo/tempo_nc2ioda.py

```diff
@@ -187,60 +197,93 @@ def _read(self):
                 exit()
 
             # clean data
-            neg_obs = obs > 0.0
-            nan_obs = obs != np.nan
+            neg_obs = ((obs > 0.0) & (err > 0.0))
+            nan_obs = ((obs != np.nan) & (err != np.nan))
```

Comment:
Do you need `isnan` here?

Reply:
Not sure if isnan would work on creating mask like here. If you think it does please provide the tested coding suggestion/example.